src/tests/conformance/messages.rs

```diff
@@ -383,79 +413,77 @@ async fn disconnects_for_trivial_issues() {
+        message_filter: MessageFilter::with_all_auto_reply().with_ping_filter(Filter::Disabled),
+        ..Default::default()
+    };
+
+    // Pong with bad nonce.
+    {
+        let mut synthetic_node = SyntheticNode::new(config.clone()).await.unwrap();
+        synthetic_node.connect(node.addr()).await.unwrap();
+
+        match synthetic_node.recv_message_timeout(TIMEOUT).await.unwrap() {
+            (_, Message::Ping(_)) => synthetic_node
+                .send_direct_message(node.addr(), Message::Pong(Nonce::default()))
+                .await
+                .unwrap(),
 
-        message => panic!("Unexpected message while waiting for Ping: {:?}", message),
+            message => panic!("Unexpected message while waiting for Ping: {:?}", message),
+        }
+
+        wait_until!(TIMEOUT, synthetic_node.num_connected() == 0);
+        synthetic_node.shut_down();
     }
-    autorespond_and_expect_disconnect(&mut stream).await;
 
-    // GetData with mixed inventory
-    let genesis_block = Block::testnet_genesis();
-    let mixed_inv = vec![genesis_block.inv_hash(), genesis_block.txs[0].inv_hash()];
-    let mut stream = initiate_handshake(node.addr()).await.unwrap();
-    Message::GetData(Inv::new(mixed_inv.clone()))
-        .write_to_stream(&mut stream)
-        .await
-        .unwrap();
-    autorespond_and_expect_disconnect(&mut stream).await;
+    // Update the filter to include ping messages.
+    let config = SyntheticNodeConfig {
+        message_filter: MessageFilter::with_all_auto_reply(),
+        ..config
+    };
+
+    // GetData with mixed inventory.
+    {
+        let synthetic_node = SyntheticNode::new(config.clone()).await.unwrap();
+        synthetic_node.connect(node.addr()).await.unwrap();
+
+        let genesis_block = Block::testnet_genesis();
+        let mixed_inv = vec![genesis_block.inv_hash(), genesis_block.txs[0].inv_hash()];
+
+        synthetic_node
+            .send_direct_message(node.addr(), Message::GetData(Inv::new(mixed_inv.clone())))
+            .await
+            .unwrap();
+
+        wait_until!(TIMEOUT, synthetic_node.num_connected() == 0);
+        synthetic_node.shut_down();
+    }
 
     // Inv with mixed inventory (using non-genesis block since all node's "should" have genesis already,
-    // which makes advertising it non-sensical)
-    let mut stream = initiate_handshake(node.addr()).await.unwrap();
-    let block_1 = Block::testnet_1();
-    let mixed_inv = vec![block_1.inv_hash(), block_1.txs[0].inv_hash()];
-    Message::Inv(Inv::new(mixed_inv))
-        .write_to_stream(&mut stream)
-        .await
-        .unwrap();
-    autorespond_and_expect_disconnect(&mut stream).await;
+    // which makes advertising it non-sensical).
+    {
+        let synthetic_node = SyntheticNode::new(config).await.unwrap();
+        synthetic_node.connect(node.addr()).await.unwrap();
 
-    // NOTE: Addr with missing timestamp cannot be run without modifying our code currently.
```

Comment:
Maybe we should consider handcrafting the bytes for such a message?

Reply:
Agreed, I'd actually consider this to be a class of "fuzz test" and applies to other messages too. We might want to remove the case for now but consider it in future.